### PR TITLE
opentabletdriver: 0.6.4.0 -> 0.6.4.0-unstable-2024-11-25

### PR DESCRIPTION
**Pull request by thiagokokada**
Going to the latest commit of the `0.6.x` branch to upgrade to .NET Framework 8.0 (since 6.0 is EOL). Not using current `main` because at last from my tests it seems broken.

Since I am already here, move to `pkgs/by-name` and format with `nixfmt-rfc-style`.

Fix issue: https://github.com/NixOS/nixpkgs/issues/360369.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by thiagokokada** [comment]
@ofborg test opentabletdriver

**Review by gepbird — CHANGES_REQUESTED** [review]
Thanks for the many fixes and cleanups!

Tested it on x86_64-linux with the NixOS module and it works well, changes mostly LGTM.

Some nits, feel free to ignore if you think it's not important:
- the desktop item shoudn't depend on `meta` attributes
- remove `with lib;` from `meta`


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 360389`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>opentabletdriver</li>
  </ul>
</details>

File: pkgs/by-name/op/opentabletdriver/package.nix [hunk 1]
```diff
@@ -94,10 +103,10 @@ buildDotnetModule rec {
 
     install -Dm644 $src/OpenTabletDriver.UX/Assets/otd.png -t $out/share/pixmaps
 
+    # Generate udev rules from source
+    export OTD_CONFIGURATIONS="$src/OpenTabletDriver.Configurations/Configurations"
```
[the remark this answers is not in the document]
Comment (thiagokokada) [reply pkgs/by-name/op/opentabletdriver/package.nix:107]: It is used to overwrite the default location for OTD configurations: https://github.com/OpenTabletDriver/OpenTabletDriver/blob/0.6.x/generate-rules.sh#L26.

[the remark this answers is not in the document]
Comment (gepbird) [reply pkgs/by-name/op/opentabletdriver/package.nix:107]: Yes, but what's the reason to overwrite it? Default location seems to work.

**Review by gepbird — APPROVED** [review]

File: pkgs/by-name/op/opentabletdriver/package.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (thiagokokada) [reply pkgs/by-name/op/opentabletdriver/package.nix:107]: Well, for one the official `default.nix` from upstream does: https://github.com/OpenTabletDriver/OpenTabletDriver/blob/master/default.nix#L77.

Also in general I prefer to not rely in magic from upstream if posible.

**Comment by thiagokokada** [comment]
@ofborg test opentabletdriver

**Comment by Hannes1909** [comment]
Just tested, looks good, updating as well as tablet and pen are working as before :) Thanks for the quick fix!

**Pull request merged**